sdk/typespec/typespec_client_core/examples/binary_data_request.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+use tokio::fs;
+use tracing::level_filters::LevelFilter;
+use tracing_subscriber::fmt::format::FmtSpan;
+use typespec_client_core::{
+    fs::FileStreamBuilder,
+    http::{Body, RequestContent},
+};
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // Log traces to stdout.
+    let _log = tracing_subscriber::fmt()
+        .with_span_events(FmtSpan::ENTER | FmtSpan::EXIT)
+        .with_max_level(LevelFilter::DEBUG)
+        .init();
+
+    // Asynchronously read the whole file into memory.
+    let body = RequestContent::from(fs::read(file!()).await?);
+    client::put_binary_data(body).await?;
+
+    // Asynchronously stream the file with the service client request.
+    #[cfg(not(target_arch = "wasm32"))]
+    {
+        let file = fs::File::open(file!()).await?;
+        let file = FileStreamBuilder::new(file)
+            // Simulate a slow, chunky request.
+            .buffer_size(256usize)
+            .build()
+            .await?;
+        let body: Body = file.into();
+        client::put_binary_data(body.into()).await?;
```

Comment:
Alternatively, maybe we need a simpler (less hops) way to get a stream into a body. We already `impl From<Box<dyn SeekableStream>> for Body` but it's not the most intuitive to make the extra hop. Maybe we support going straight to a `RequestContent<T>`?

Note: ignore the `buffer_size()` call. It's to simulate a slow (small buffer as a proxy) client request only.